app/src/lib/remote-parsing.ts

```diff
@@ -35,26 +37,42 @@ export function parseRemote(url: string): IGitRemoteURL | null {
       continue
     }
 
-    const hostname = result[1]
-    const owner = result[2]
-    const name = result[3]
-    if (hostname) {
-      return { hostname, owner, name }
+    if (result.length === 4) {
+      const hostname = result[1]
+      const owner = result[2]
+      const name = result[3]
+      if (hostname) {
+        return { hostname, owner, name }
+      }
+    }
+
+    if (result.length === 3) {
```

Comment:
I'm not a fan of only determining this based on the number of capture groups in the regex. It feels fragile and prone to problems in the future if we alter any of these regexes. Could we split this particular regex out of the list and deal with it specifically when no other regexes match?